client/apps/edit/components/content/index.coffee

```diff
@@ -0,0 +1,46 @@
+React = require 'react'
+moment = require 'moment'
+_ = require 'underscore'
+HeroSection = React.createFactory require '../hero_section/index.coffee'
+HeaderSection = React.createFactory require '../section_header/index.coffee'
+SectionList = React.createFactory require '../section_list/index.coffee'
+{ div, p, textarea } = React.DOM
+
+
+module.exports = React.createClass
+  displayName: 'EditContent'
+
+  componentWillMount: ->
+    @debouncedSave = _.debounce((->
+      @props.article.save()
+      @forceUpdate()
+    ), 800)
+    @props.article.sections.on 'change add remove reset', => @saveArticle()
+    @props.article.heroSection.on 'change remove', => @saveArticle()
+
+  saveArticle: ->
+    unless @props.article.get('published')
+      $('#edit-save').addClass('is-saving')
```

Comment:
FYI this is the kind of thing that could be further refactored into React. e.g. the `is-saving` CSS class is essentially a piece of state. So one could express this like...

````coffeescript
saveArticle: ->
  unless @props.article.get('published')
    @setState isSaving: true

# ... in the header component somewhere
render: ->
  # ..
  a { id: 'edit-save', class: if @state.isSaving then 'is-saving' else '' }
  # ...
````

Not for this PR because that would involve refactoring the header into React too, but food for thought.